lucene/core/src/java/org/apache/lucene/util/automaton/Operations.java

```diff
@@ -1091,25 +1091,33 @@ public static String getCommonPrefix(Automaton a) {
    * @return common prefix, which can be an empty (length 0) BytesRef (never null)
    */
   public static BytesRef getCommonPrefixBytesRef(Automaton a) {
```

Comment:
Lazy create structures (WildcardQuery may not have a first single-path state), and only one visited.add() call. This speeds up mainly WildcardQuery (a couple % in my luceneutil measures).

Reply:
I don't think this is the right tradeoff. It makes the code more complex, saving the cost of creating a few simple ordinary objects. I hate to say i don't trust your benchmark, but I don't trust your benchmark represents a typical case here. We should keep this code simple, there are other ways it can be improved.

Reply:
Ok, removed.